test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -86,6 +121,97 @@ TEST_F(XdsOverrideHostTest, NoConfigReportsError) {
       absl::InvalidArgumentError("Missing policy config"));
 }
 
+TEST_F(XdsOverrideHostTest, OverrideHost) {
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto maybe_picker = InitRoundRobin(kAddresses);
+  ASSERT_TRUE(maybe_picker.has_value());
+  auto picker = *maybe_picker;
+  ASSERT_NE(picker, nullptr);
+  // Check that the host is overridden
+  std::map<UniqueTypeName, std::string> call_attributes{
+      {XdsHostOverrideTypeName(), "127.0.0.1:442"}};
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[1]);
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[1]);
+  call_attributes[XdsHostOverrideTypeName()] = std::string("127.0.0.1:441");
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[0]);
+  EXPECT_EQ(ExpectPickComplete(picker.get(), call_attributes), kAddresses[0]);
+}
+
+TEST_F(XdsOverrideHostTest, OverrideHostSubchannelNotFound) {
+  // Send address list to LB policy.
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto maybe_picker = InitRoundRobin(kAddresses);
+  ASSERT_TRUE(maybe_picker.has_value());
+  auto picker = *maybe_picker;
+  ASSERT_NE(picker, nullptr);
+  std::map<UniqueTypeName, std::string> call_attributes{
+      {XdsHostOverrideTypeName(), "no such host"}};
+  std::unordered_set<std::string> picks;
+  for (size_t i = 0; i < kAddresses.size(); ++i) {
+    auto pick = ExpectPickComplete(picker.get(), call_attributes);
+    ASSERT_TRUE(pick.has_value());
+    picks.emplace(*pick);
+  }
+  ASSERT_THAT(picks, UnorderedElementsAreArray(kAddresses));
+}
+
+TEST_F(XdsOverrideHostTest, SubchannelsComeAndGo) {
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  auto maybe_picker = InitRoundRobin(kAddresses);
+  ASSERT_TRUE(maybe_picker.has_value());
+  auto picker = *maybe_picker;
+  ASSERT_NE(picker, nullptr);
+  // Make sure child policy works
+  EXPECT_NE(ExpectPickComplete(picker.get()), ExpectPickComplete(picker.get()));
```

Comment:
Is this expectation actually passing?  It seems like it shouldn't be, because we're using round_robin here, so there should not be two successive picks for the same address.

In any case, I don't think this check is actually needed, since `InitRoundRobin()` is already checking the initial picker behavior for us.